### DISCUSSION TITLE
Help with Proscan Format

### DISCUSSION THREAD

**Discussion by CSKuhn** [chuot/rdio-scanner#192] [General]
Trying to get Proscan mp3 files to work in Dirwatch. Have tried many formats in each program without total success. Can someone show me the mask format they use in Rdio Scanner and the file name format they use in Proscan. Thanks!

**Comment by chuot** [comment]
Try this on ProScan for custom filename format `%YY-%M-%DD_%TT_%S_%G_%C_%FT`.

And this mask for dirwatch `#DATE_#TIME_#SYSLBL_#GROUP_#TGLBL_#TGMHZ` with extension set to `mp3`.

**Reply by CSKuhn** [reply]
Thank you for the quick reply. I updated both programs with the strings and I am still getting "dirwatch.ingest: no talkgroup" like I was getting before. When I hardcode (select from dropdown) the TG and use the #SYS metatag setting for System, I get "dirwatch.ingest: no system". So the only way to make it work is to pick a TG and SYS manually. But then I do not see the actual groups, tgs, etc on the display - but the files are processed and passed.

Here is an example of the Proscan filename "2022-05-26_1-44-28 PM_Washington County_Washington County Fire_FD Fire Main_13463.mp3". Have never been able to see the freq, so I have tried changing the rdio mask to #TG with no change to TG error. That number on the end is the actual TG.

Attached is a screenshot
![Screenshot 2022-05-26 141217](https://user-images.githubusercontent.com/106329687/170550148-fc69c1d0-d18c-4ca6-b606-70678d7d8440.jpg)
.

**Comment by chuot** [comment]
OK, I see.

Dirwatch only works with local files. Any remote folder mounted locally over the network will not work. This is because polling mode is not supported by the [library](https://github.com/fsnotify/fsnotify) we use, and is inefficient anyway.

You can install a new Rdio Scanner instance on the same PC that has ProScan running on to ingest files with dirwatch. Then configure a downstream on that same instance to forward ingested files to your main instance, which is the one that should be available to the outside world.

**Reply by CSKuhn** [reply]
Thanks! I setup on the same machine as Proscan and pointed to local files. Still the same issue.

My main system is a SDRTrunk monitoring 2 large county systems and works great. Did not want to add another county so was using an SDS200 scanner and Proscan. Oh well, just added the system to SDRTrunk and everything is fine and appears to be working with 4 tunners. Might add a couple more tunners to play safe :)

**Comment by Thurmont** [comment]
chuot says "Dirwatch only works with local files".... So is the "G: Drive" Google creates using it's Backup/Sync Windows application considered a local drive or a network drive?

 Reason I ask is I currently have ProScan save all the mp3 recordings to a folder on this G Drive which then gets sync'ed to Google Drive. I've shared this Google Drive folder with a couple friends so they have access to the local PD and Fire recordings too. I'm hoping I can get rdio-scanner to recognize these files and use them within it's app on my cellphone to make browsing and listening to them easier.

**Comment by chuot** [comment]
I can't say for sure with Google Drive, but if the files aren't picked up by dirwatch, they're probably not local. This should work if Rdio Scanner is installed on the same PC as ProScan and is monitoring the recording folder. As a test, try eliminating Google Drive from the equation to see if that works.

**Reply by Thurmont** [reply]
I think my Google Drive folder is working since I can get rdio-scanner (sometimes) to see a "system" sometimes but then it complains about not having a talkgroup. Other times, it complains about not having either. Both ProScan and rdio-scanner are running on the same PC.

 I'm currently using a Uniden BCD536HP to receive, ProScan to record, and Zello to stream. I want to have rdio-scanner in the mix to organize and playback but now I'm trying to sort out how ProScan and rdio-scanner can work together. I think my troubles involve setting up ProScan in a way so that the filenames it creates can be parsed out by rdio-scanner.

**Reply by bobfeser** [reply]
@Thurmont did you ever figure this out?? I am dealing with the same issue with proscan/rdio-scanner

**Comment by adl5007** [comment]
I came here looking for a solution to this exact issue.   Rdio-scanner is ingesting the file but I’m getting the similar “dirwatch.ingest: no talkgroup” message.  The only workaround is to specify the talkgroup manually.

**Reply by adl5007** [reply]
I was actually able to get everything working - not fully sure what the issue was, but for those interested here are the masks I am using that now work:
ProScan `%YY-%M-%DD_%TT_%S_%C_%F`
Rdio Scanner `#DATE_#TIME_#SYSLBL_#TGLBL_#TGMHZ`

I have a new issue now I'm trying to work through - it seems that Rdio-scanner is ingesting the ProScan recording as soon as the file is created and while the transmission may still be ongoing.  And then once the transmission/recording is complete it ingests the file again.  As a result, I get a call with the first second or so of audio in Rdio Scanner, and then a second call with the full audio.  At first Rdio Scanner was not ingesting at full call thinking it was a duplicate, so I had to turn off duplicate detection.

Has anyone come across this before?

**Reply by bobfeser** [reply]
Did you have to create a different Dirwatch for each system that you create? Also for the proscan mask that was the file name?

**Reply by noamlivne** [reply]
I am also suffering from the duplicate/triple call recordings from Proscan to Rdioscanner. It is as if Rdioscanner tries to digest the recordings before they finish and creates more than one call that is cut in the middle.
I hope there will be a solution for this issue.

**Reply by jamie75** [reply]
I know this is old, just started playing with this software and getting this situation. Did you ever find a solution for this?

**Reply by noamlivne** [reply]
I did not succeed with Proscan -> RdioScanner. 
So what I did is Proscan -> Trunking Recorder -> RdioScanner.
I followed the instructions of how to "connect" Trunking Recorder to Proscan (https://www.scannerbox.us/TrunkingRecorder/ProScan/).
Trunking Recorder Imports the Calls from Proscan.
Trunking Recorder Uploads the Calls to RdioScanner using the API Key method.
That worked for me. Without any masks needed.
Follow the additional important instructions at https://www.scannerbox.us/TrunkingRecorder/CallUploads/

**Comment by Brantlab** [comment]
I have Proscan and have it working pulling files in but am struggling with groups. Here is an example of a file name.
`dirwatch.ingest: no talkgroup, C:\Recordings\2023-03-02_13-59-44_MARCS_VWCPD-1_772.181250_47532_VanWertCounty81.mp3`

The TGID is 47532. If I remove the group name off the end it works fine.... 
Here is my ingest file setting : `#DATE_#TIME_#SYSLBL_#TGLBL_#TGMHZ_#TG_#GROUP`
any ideas?
